cli/ssh_test.go

```diff
@@ -159,9 +122,17 @@ func TestSSH(t *testing.T) {
 			t.Skip("Windows doesn't seem to clean up the process, maybe #7100 will fix it")
 		}
 
-		client, workspace, agentToken := setupWorkspaceForAgent(t, nil)
+		store, ps := dbtestutil.NewDB(t)
+		client := coderdtest.New(t, &coderdtest.Options{Pubsub: ps, Database: store})
+		client.SetLogger(slogtest.Make(t, nil).Named("client").Leveled(slog.LevelDebug))
+		first := coderdtest.CreateFirstUser(t, client)
+		userClient, user := coderdtest.CreateAnotherUser(t, client, first.OrganizationID)
+		workspace, agentToken := dbfake.WorkspaceWithAgent(t, store, database.Workspace{
+			OrganizationID: first.OrganizationID,
+			OwnerID:        user.ID,
+		})
```

Comment:
Unless I'm making the wrong assumption for this change/code, maybe `setupWorkspaceForAgent` could return a struct instead, with store/ps so we could avoid needing this ceremony when we require those. Just a thought.

Reply:
We could.  I decided not to since there are only two places that require it right now.